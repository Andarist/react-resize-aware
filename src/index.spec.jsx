import React from 'react';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import ResizeAware from './index.js';

function DummyComponent({width, height}) {
  return <div>{width}x{height}</div>;
}

function DummyComponent2({getRef, ...props}) {
  return <div ref={getRef} {...props} />;
}

it('allows to use its children as target', () => {
  const wrapper = mount(
    <ResizeAware style={{position: 'relative'}}>
      <DummyComponent />
    </ResizeAware>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('allows to define an `onResize` property', () => {
  const handleResize = jest.fn();
  const wrapper = mount(
    <ResizeAware style={{position: 'relative'}} onResize={handleResize} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('allows to define a custom component', () => {
  const wrapper = shallow(
    <ResizeAware style={{position: 'relative'}} component={DummyComponent} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
