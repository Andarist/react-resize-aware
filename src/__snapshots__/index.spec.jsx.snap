// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows to define a custom component 1`] = `
<DummyComponent
  getRef={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <object
    onLoad={[Function]}
    style={
      Object {
        "display": "block",
        "height": "100%",
        "left": 0,
        "overflow": "hidden",
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": -1,
      }
    }
    type="text/html"
  />
</DummyComponent>
`;

exports[`allows to define an \`onResize\` property 1`] = `
<ResizeAware
  component="div"
  onResize={[Function]}
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <object
      onLoad={[Function]}
      style={
        Object {
          "display": "block",
          "height": "100%",
          "left": 0,
          "overflow": "hidden",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": -1,
        }
      }
      type="text/html"
    />
  </div>
</ResizeAware>
`;

exports[`allows to use its child as target 1`] = `
<ResizeAware
  component="div"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <object
      onLoad={[Function]}
      style={
        Object {
          "display": "block",
          "height": "100%",
          "left": 0,
          "overflow": "hidden",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": -1,
        }
      }
      type="text/html"
    />
    <DummyComponent>
      <div>
        x
      </div>
    </DummyComponent>
  </div>
</ResizeAware>
`;

exports[`allows to use its children as target 1`] = `
<ResizeAware
  component="div"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <object
      onLoad={[Function]}
      style={
        Object {
          "display": "block",
          "height": "100%",
          "left": 0,
          "overflow": "hidden",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "width": "100%",
          "zIndex": -1,
        }
      }
      type="text/html"
    />
    <DummyComponent>
      <div>
        x
      </div>
    </DummyComponent>
  </div>
</ResizeAware>
`;
